docs/content/develop/test/test.md

```diff
@@ -222,6 +222,17 @@ An update test is a test that creates the target resource and then makes updates
 {{< /tab >}}
 {{< /tabs >}}
 
+## Add unit tests
+
+A unit test verifies functionality that is not related to interactions with the API, such as
+[diff suppress functions]({{<ref "/develop/field-reference#diff_suppress_func" >}})),
+[validation functions]({{<ref "/develop/field-reference#validation" >}}),
+CustomizeDiff functions, and so on.
+
+Unit tests should be added to the appropriate folder in [`magic-modules/mmv1/third_party/terraform/services`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/services) in the file called `resource_PRODUCT_RESOURCE_test.go`. (You may need to create this file if it does not already exist. Replace PRODUCT with the product name and RESOURCE with the resource name; it should match the name of the generated resource file.)
+
+Unit tests should be named like `TestFunctionName` - for example, `TestDiskImageDiffSuppress` tests the `DiskImageDiffSuppress` function.
```

Comment:
I think it would be useful to link an example here. Not necessarily comprehensively explain every type of unit test, but to illustrate the way tests that seek to verify (expected) error states, or other plan diffs. We currently lack explanation of these specifics in our documentation for how to make tests. Or at least add a link to https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests/teststep which goes into a number of the assertions more specifically.

Reply:
Added an example - let me know what you think! I'm not linking to the acceptance tests doc because these are unit tests rather than acceptance tests.

Reply:
<img width="749" alt="Screenshot 2024-04-17 at 2 28 04 PM" src="https://github.com/GoogleCloudPlatform/magic-modules/assets/299979/7dfb6fa4-9a69-46b6-84cc-eaa721a59531">